src/librustc_mir/interpret/memory.rs

```diff
@@ -580,10 +580,9 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
                 let layout = self.tcx.layout_of(ParamEnv::empty().and(ty)).unwrap();
                 Ok((layout.size, layout.align.abi))
             }
-            Some(GlobalAlloc::Memory(alloc)) =>
-            // Need to duplicate the logic here, because the global allocations have
-            // different associated types than the interpreter-local ones.
-            {
+            Some(GlobalAlloc::Memory(alloc)) => {
+                // Need to duplicate the logic here, because the global allocations have
```

Comment:
I usually move such comments before the pattern if that will allow the pattern and the arm body to be on one line

Reply:
The comment belongs to the body of the arm though, not the pattern.

Reply:
Sure, but there can be no confusion here about what is meant

Reply:
But here the code should not be on the same line as the pattern anyway, as the other arms of the same match are much longer. Making them both on the same line makes it way too easy to miss the fact that there even is a branch here.

IMO, `pat => expr` should only be used if i can be used consistently for all arms of this `match`. Otherwise, getting consistent visual cluse about where each arm starts and ends is more important than saving vertical space.

Reply:
The next arm is on a single line, too. I don't have a strong opinion though, so marking as resolved

Reply:
Yeah I'd like that to be in its own line as well but it's impossible. :(